src/py42/services/orgs.py

```diff
@@ -232,3 +260,13 @@ def update_settings(self, org_settings):
             org_response=org_response,
             org_settings_response=org_settings_response,
         )
+
+    def _get_guid_by_id(self, org_id, id_key="orgId"):
```

Comment:
I wonder if we should cache the results of this method, so we're not looping through every org for every individual org interaction.

Reply:
Thats a good idea,  so like every time you call it, it stores an entry of `{"orgId": "orgGuid"}` in a prop like we did on watchlists? (seems like the Uid's are used rarely enough that that caching wouldn't be super necessary)